xonsh/procs/executables.py

```diff
@@ -57,10 +74,20 @@ def is_executable_in_windows(filepath, env=None):
         return False
 
 
-def is_executable_in_posix(filepath):
-    """Check the file is executable in POSIX."""
+def is_executable_in_posix(filepath, check_file_exist=True):
+    """Check the file is executable in POSIX.
+
+    Parameters
+    ----------
+    filepath : str
+        Path to file.
+    check_file_exist : bool
+        If ``False`` do not check that file exists. This made to consistency with ``is_executable_in_windows``.
+    """
     try:
-        return filepath.is_file() and os.access(filepath, os.X_OK)
```

Comment:
it will be enough to check the access? why do we need to check is_file and access?

Reply:
Ok it seems, the dir can have X_OK